core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java

```diff
@@ -876,11 +892,9 @@ private ConnectionListener createConnectionEventListener(Subject subject, Connec
    }
 
    /**
-    * Destroy a connection
-    *
-    * @param cl the connection to destroy
+    * {@inheritDoc}
     */
-   void doDestroy(ConnectionListener cl)
+   public void connectionListenerDestroyed(ConnectionListener cl)
```

Comment:
Only the statistics update should remain in the connectionListenerDestroyed method. The remainder is the part that we moved to inside ConnectionListener.destroy.